src/sqlfluff/core/rules/analysis/select_crawler.py

```diff
@@ -246,10 +247,11 @@ def finish_segment():
             if event == "start":
                 # "start" means we're starting to process a new segment.
                 if path[-1].is_type(
-                    "set_expression",
+                    "merge_statement",
                     "select_statement",
-                    "values_clause",
+                    "set_expression",
                     "update_statement",
+                    "values_clause",
```

Comment:
I added `merge_statement`. The other "changes" were just me sorting the types for easier reading.